libs/utils/src/crashsafe_dir.rs

```diff
@@ -92,6 +84,44 @@ pub fn path_with_suffix_extension(original_path: impl AsRef<Path>, suffix: &str)
         .with_extension(new_extension.as_ref())
 }
 
+pub fn fsync_file_and_parent(file_path: &Path) -> io::Result<()> {
+    let parent = file_path.parent().ok_or_else(|| {
+        io::Error::new(
+            io::ErrorKind::Other,
+            format!("File {} has no parent", file_path.display()),
+        )
+    })?;
+
+    fsync(file_path)?;
+    fsync(parent)?;
+    Ok(())
+}
+
+pub fn fsync(path: &Path) -> io::Result<()> {
```

Comment:
This seems really verbose. What was wrong with just `File::open(path)?.sync_all()?` ? Don't the io::Errors contain enough context?

Reply:
```rust
// in the upper mod
pub fn old_fsync(path: &Path) -> io::Result<()> {
    let file = File::open(path)?;
    file.sync_all()?;
    Ok(())
}

#[test]
    fn fsync_test() {
        println!("{:#?}", super::fsync(Path::new("bvsbsbs")));
        println!("{:#?}", super::old_fsync(Path::new("bvsbsbs")));
    }
```

and the verbose (and a bit adjusted) version will produce this output:

```
Err(
    Custom {
        kind: Other,
        error: "Failed to open the file \"bvsbsbs\": No such file or directory (os error 2)",
    },
)
Err(
    Os {
        code: 2,
        kind: NotFound,
        message: "No such file or directory",
    },
)
```

I always have issues with some io::Error and ? due to zero clue what exactly is wrong, which file is actually missing, etc.